webull.py

```diff
@@ -644,14 +649,30 @@ def get_bars(self, stock=None, tId = None, interval='m1', count=1, extendTrading
         return df.iloc[::-1]
 
 
-    def get_calendar(self):
+    def get_calendar(self,stock=None, tId=None):
+        """
+        There doesn't seem to be a way to get the times the market is open outside of the charts.
+        So, best way to tell if the market is open is to pass in a popular stock like AAPL then
+        and see the open and close hours are as would be marked on the chart
+        and see if the last trade date is the same day as today's date
+        :param stock:
+        :param tId:
+        :return: dict of 'market open', 'market close', 'last trade date'
+        """
+        if not tId is None:
+            pass
+        elif not stock is None:
+            tId = self.get_ticker(stock)
+        else:
+            raise ValueError('Must provide a stock symbol or a stock id')
+
         params = {'type': 'm1', 'count': 1, 'extendTrading': 0}
-        response = requests.get(urls.bars(self.get_ticker('AAPL')), params=params)
+        response = requests.get(urls.bars(tId), params=params)
         data = response.json()
         last_trade_date = datetime.fromtimestamp(int(data[0]['data'][0].split(',')[0]))
         for d in data[0]['dates']:
             if d['type'] == 'T':
-                return {'start':d['start'],  'end':d['end'], 'last trade':last_trade_date}
+                return {'market open':d['start'],  'market close':d['end'], 'last trade date':last_trade_date}
```

Comment:
It appears that data comes back in local timezone(Central), and dates comes back in Eastern (probably my weird acct configuration), is not consistent for providing back to caller.

It'd be nice to just have a 4th dict entry that was 'isOpen': True / False
something like 
True if today is on last_trade_date and >= open and < close else False

[{'data': ['1581714000,324.83,324.95,324.95,324.68,324.82,2004688,324.65'],
  'dates': [{'avgShow': 1,
             'end': '16:00:00',
             'start': '09:31:00',
             'type': 'T'}],
  'exchangeStatus': False,
  'hasMore': 1,
  'preClose': '324.82',
  'tickerId': 913256135,
  'timeZone': 'America/New_York'}]
1581714000 => 2020-02-14 15:00:00

FWIW: I wouldn't recommend using webull as the primary source for stock data. I like ally, alpaca, and iex for what I need. I know alpaca REST class has a get_clock function that users can just check .is_open (https://docs.alpaca.markets/api-documentation/api-v2/clock/)

Reply:
Just trying to make it complete...  I noticed time zone wasn't being added to the get_bars...  but it is a item in the response dict, so I will work on adding pytz to the mix. I am in eastern/nyc timezone, so it all looks the same to me. I love alpaca, but they don't want to do cash accts, so playing with webull now